docs/api/utils/combineKeys.md

```diff
@@ -22,6 +22,66 @@ export const combineKeys = <K, T>(
 
 A stream with a map containing the latest value from the stream of each key.
 
+### Examples
+
+You're typically going to call this after `partitionByKey()`:
+
+```ts
+const [getByKey, keys$] = partitionByKey(source$, x => x.key);
+const combined$ = combineKeys(keys$, getByKey);
+```
+
+Here it is in action, picking up from the [`partitionByKey()`](partitionByKey)
+example. You have a stream representing the list of pets, you have a function
+that returns the stream associated with each pet, and now combineKeys() gives
+you a stream that emits a map of this whole data structure whenever any of it
+changes. If you're careful about where you bind these you can save a lot of
+component updates:
+
+```ts
+interface Pet {
+  id: number;
+  pet: string,
+  pos: number;
+}
+
+const petNames = ["Fluffy", "Bella", "Nala", "Nocturne", "Teddy"];
+
+let [petRace$, petRaceDispatch] = createSignal<Pet>();
+
+petRace$ = petRace$.pipe(startWith(
+  ...petNames.map((pet, id): Pet => ({pet, id, pos: 1})),
+));
+
+const [petByID, pets$] = partitionByKey(petRace$, x => x.id)
+const keyMap$ = combineKeys(pets$, petByID);
+
+const leadingPet$ = keyMap$.pipe(map(x => // map to pet with highest pos
+  Array.from(x.entries())
+    .sort(([k1,v1], [k2,v2]) => v2.pos - v1.pos)[0][1]
+));
+
+const advancingPet$: Observable<Pet> = interval(1000).pipe(
+  withLatestFrom(keyMap$),
+  map(([_, x]) => x),
+  takeWhile(x => {
+    for (let [k,v] of x) {
+      if (v.pos == 20) return false  // win condition
+    }
+    return true;
+  }),
+  map((x: Map<number, Pet>) =>
+    x.get(Math.floor(Math.random() * x.size)) as Pet),
+  map(pet => ({...pet, pos: pet.pos + 1})),  // increment position
+  tap(petRaceDispatch),
+);
```

Comment:
This feels a bit dangerous 😅 It's cool because this is an example that it has a purpose, but it's more like a side effect rather than a state by itself.

I don't think I would use `bind` on it - Instead I would use a top-level subscription or even better, reference it directly on a `<Subscribe>` component.

I would probably leave it as is, because on every project there are side-effecty features which are like these and it's a nice example, but I would remove the `bind` down below.

Reply:
Okay no problem. I called bind on it so in the demo you could display the last pet to have advanced (the pet race is more exciting that way :) I also had a 'Start race' button that subscribes to this but the thing was getting too long for a brief example and started turning into a demo.